src/esp/sensor/VisualSensor.cpp

```diff
@@ -3,17 +3,68 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "VisualSensor.h"
+#include <Magnum/EigenIntegration/Integration.h>
 
 #include <utility>
 
 #include "esp/gfx/RenderTarget.h"
 
 namespace esp {
 namespace sensor {
-VisualSensor::VisualSensor(scene::SceneNode& node, SensorSpec::ptr spec)
-    : Sensor{node, std::move(spec)}, tgt_{nullptr} {}
 
-VisualSensor::~VisualSensor() = default;
+VisualSensorSpec::VisualSensorSpec() : SensorSpec() {
+  sensorType = SensorType::Color;
+}
+
+void VisualSensorSpec::sanityCheck() {
+  SensorSpec::sanityCheck();
+  bool isVisualSensor =
+      (sensorType == SensorType::Color || sensorType == SensorType::Depth ||
+       sensorType == SensorType::Normal || sensorType == SensorType::Semantic);
+  CORRADE_ASSERT(
+      isVisualSensor,
+      "VisualSensorSpec::sanityCheck(): sensorType must be Color, Depth, "
+      "Normal, or Semantic", );
+  if (noiseModel == "Gaussian" || noiseModel == "Poisson" ||
```

Comment:
@Skylion007 : is noise model a must have??

Reply:
I am not sure if noise model is a must have, but I see in hab-lab that there are sensors which use the noise model (seems like only visual sensors though).  @Skylion007 what do you think we should do here? 

These checks in the VisualSensorSpec mirror checks on they python side.

Reply:
Yili, I think your question is whether we want noiseModel to be an attribute of SensorSpec or VisualSensorSpec, (and not to get rid of noiseModel in all SensorSpecs) right? Since for nonVisualSensorSpecs noiseModel will always be None. Let me know if this is what you're asking.

Reply:
Ok, from discussion in hab-sim dev sync we want to keep the attribute in SensorSpec because we anticipate that future non-visual sensor specs will want to add a noiseModel

Reply:
I think we should keep the checks for noiseModel in SanityCheck because it would be nice to have for C++ and js, but let me know if you would like them removed because of redundancy on the python side.

Reply:
All the noise models are in Pure Python now, it would be a good idea to move them to C++ if performance warrants it though.